src/main/java/org/generationcp/middleware/service/impl/inventory/LotServiceImpl.java

```diff
@@ -25,12 +28,33 @@ public LotServiceImpl(final HibernateSessionProvider sessionProvider) {
 	}
 
 	@Override
-	public List<LotDto> searchLots(final LotsSearchDto lotsSearchDto,final Pageable pageable) {
+	public List<ExtendedLotDto> searchLots(final LotsSearchDto lotsSearchDto,final Pageable pageable) {
 		return this.daoFactory.getLotDao().searchLots(lotsSearchDto, pageable);
 	}
 
 	@Override
 	public long countSearchLots(final LotsSearchDto lotsSearchDto) {
 		return this.daoFactory.getLotDao().countSearchLots(lotsSearchDto);
 	}
+
+	@Override
+	public Integer saveLot(final LotGeneratorInputDto lotDto) {
+		final String stockId = lotDto.getStockId();
+
+		final Lot lot = new Lot();
+		lot.setUserId(lotDto.getUserId());
+		lot.setComments(lotDto.getComments());
+		lot.setCreatedDate(new Date());
+		lot.setEntityId(lotDto.getGid());
+		lot.setEntityType("GERMPLSM");
+		lot.setLocationId(lotDto.getLocationId());
+		lot.setStockId(stockId);
+		lot.setStatus(0);
+		lot.setSource(0);
```

Comment:
@vmaletta not so sure about 0 being the source in all cases, we should be careful in the near future to set the right value here